x/accreditations/internal/types/msgs.go

```diff
@@ -0,0 +1,207 @@
+// specific reward to an accrediter verifying that he has previously
+// accreditated all the users.
+// Note that the signer should be a trustworthy one in order to avoid
+// unauthorized reward distributions.
+type MsgDistributeReward struct {
+	Accrediter sdk.AccAddress `json:"accrediter"`
+	User       sdk.AccAddress `json:"user"`
+	Signer     sdk.AccAddress `json:"signer"`
+	Reward     sdk.Coins      `json:"reward"`
+}
+
+func NewMsgDistributeReward(accrediter sdk.AccAddress, reward sdk.Coins, user, signer sdk.AccAddress) MsgDistributeReward {
+	return MsgDistributeReward{
+		Accrediter: accrediter,
+		User:       user,
+		Signer:     signer,
+		Reward:     reward,
+	}
+}
+
+// Route Implements Msg.
+func (msg MsgDistributeReward) Route() string { return RouterKey }
+
+// Type Implements Msg.
+func (msg MsgDistributeReward) Type() string { return MsgTypeDistributeReward }
+
+// ValidateBasic Implements Msg.
+func (msg MsgDistributeReward) ValidateBasic() sdk.Error {
+	if msg.Accrediter.Empty() {
+		return sdk.ErrInvalidAddress(msg.Accrediter.String())
+	}
+	if msg.User.Empty() {
+		return sdk.ErrInvalidAddress(msg.User.String())
+	}
+	if msg.Signer.Empty() {
+		return sdk.ErrInvalidAddress(msg.Signer.String())
+	}
+	if msg.Reward.Empty() {
+		return sdk.ErrUnknownRequest("reward cannot be empty")
+	}
+	if msg.Reward.IsAnyNegative() {
+		return sdk.ErrUnknownRequest("rewards cannot be negative")
+	}
+	return nil
+}
+
+// GetSignBytes Implements Msg.
+func (msg MsgDistributeReward) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
+
+// GetSigners Implements Msg.
+func (msg MsgDistributeReward) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Signer}
+}
+
+// --------------------------------
+// --- MsgDepositIntoLiquidityPool
+// --------------------------------
+
+// MsgDepositIntoLiquidityPool should be used when wanting to deposit a specific
+// amount into the liquidity pool which contains the total amount of rewards to
+// be distributed upon an accreditation
+type MsgDepositIntoLiquidityPool struct {
+	Depositor sdk.AccAddress `json:"depositor"`
+	Amount    sdk.Coins      `json:"amount"`
+}
+
+func NewMsgDepositIntoLiquidityPool(amount sdk.Coins, depositor sdk.AccAddress) MsgDepositIntoLiquidityPool {
+	return MsgDepositIntoLiquidityPool{
+		Depositor: depositor,
+		Amount:    amount,
+	}
+}
+
+// Route Implements Msg.
+func (msg MsgDepositIntoLiquidityPool) Route() string { return RouterKey }
+
+// Type Implements Msg.
+func (msg MsgDepositIntoLiquidityPool) Type() string { return MsgTypesDepositIntoLiquidityPool }
+
+// ValidateBasic Implements Msg.
+func (msg MsgDepositIntoLiquidityPool) ValidateBasic() sdk.Error {
+	if msg.Depositor.Empty() {
+		return sdk.ErrInvalidAddress(msg.Depositor.String())
+	}
+	if msg.Amount.Empty() {
+		return sdk.ErrUnknownRequest("amount cannot be empty")
+	}
+	if msg.Amount.IsAnyNegative() {
```

Comment:
Also checked in `keeper.go`

Reply:
Please read [here](https://github.com/commercionetwork/commercionetwork/pull/24#discussion_r325691910)